hls4ml/templates/coyote_accelerator/host_libs.cpp

```diff
@@ -39,9 +39,7 @@ void CoyoteInference::predict() {
     }
 
     // Poll on completion; each batch increments the counter by one
-    while (coyote_thread.checkCompleted(coyote::CoyoteOper::LOCAL_TRANSFER) != batch_size) {
-        std::this_thread::sleep_for(std::chrono::nanoseconds(50));
-    }
+    while (coyote_thread.checkCompleted(coyote::CoyoteOper::LOCAL_TRANSFER) != batch_size) {}
```

Comment:
Wouldn't this cause 100% CPU usage while the program is polling?

Reply:
On one of the cores, yes. 

But sleep for less than 50us is not well-defined on most Linux platforms. Hence, measured latency can go from ~4us to >50us even though the "true" execution latency is still 4us.